File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target)
    {
        HashMap<Integer,Integer> hmap=new HashMap<>();
        int arr[]=new int[2];
        int curr=0;
        for(int i=0;i<nums.length;i++)
        {
            curr=target-nums[i];
            if(hmap.containsKey(curr))
            {
                arr[0]=hmap.get(curr);
                arr[1]=i;
                return arr;
            }
            hmap.put(nums[i],i);
        }
        return arr;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n=len(nums)
        numMap={}
        for i in range(n):
            numMap[nums[i]]=i
        
        for i in range(n):
            comp= target - nums[i]
            if comp in numMap and numMap[comp]!=i:
                return [i,numMap[comp]]
        return []